lyngk_board.js

```diff
@@ -0,0 +1,210 @@
+  };
+  var colorArray = ['R', 'K', 'B', 'I', 'G'];
+  var kTestString1 = 'KRI|-|G|-|GR|-|IG|KJ|-|-|-|-|RK|-|J|KB|I|-|-|-|-|K|G|-|I|GBI|-|I|GIJR|B|-|RB|RKBI|G|RK|-|GB|-|-|RB|-|BK|-';
+
+  var kTestString = 'G|B|K|K|R|K|I|I|K|G|J|K|K|B|I|B|G|R|G|K|R|G|K|J|B|R|R|B|G|J|G|I|G|B|R|B|I|I|R|B|I|R|I|';
+  my.startUp = drawBoard(context);
+
+  function drawBoard(ctx) {
+    ctx.clearRect(0, 0, c.width, c.height);
+    var colorCount = {
+      Ivory: 8,
+      Blue: 8,
+      Red: 8,
+      Green: 8,
+      Black: 8,
+      White: 3
+    };
+    
+    for (var i = 0; i < verticalX.length; i++) {
+      drawLine((xRoot * verticalX[i] + padding), (yOffset * verticalY1[i] + padding), (xRoot * verticalX[i] + padding), (yOffset * verticalY2[i] + padding), ctx);
+    }
+    for (var j = 0; j < diagonalDestination.length; j++) {
+      drawLine((xRoot * diagonalSourceCoords[j][0] + padding), (yOffset * diagonalSourceCoords[j][1] + padding), (xRoot * diagonalDestination[j][0] + padding), (yOffset * diagonalDestination[j][1] + padding), ctx);
+    }
+    var boardString = document.getElementById('boardstring').innerHTML;
+    translateBoardString(boardString, context);
+    populateClaimedCanvases();
+  }
+
+  function drawLine(x1, y1, x2, y2, ctx) {
+    ctx.beginPath();
+    ctx.moveTo(x1, y1);
+    ctx.lineTo(x2, y2);
+    ctx.stroke();
+  }
+
+  function drawGamePiece(xcenter, ycenter, color, ctx) {
+    ctx.beginPath();
+    ctx.arc(xcenter, ycenter, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[color];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function drawUnclaimedPiece(color, piecePosition) {
+    var board =document.getElementById("unclaimedCanvas");
+    var ctx=board.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(40, piecePosition * 62, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function drawClaimedPiece(color, boardNumber, piecePosition) {
+    boardLabel = '';
+    switch(boardNumber) {
+      case 1:
+        boardLabel = "player1canvas";
+        break;
+      case 2:
+        boardLabel = "player2canvas";
+        break;
+    }
+    var board =document.getElementById(boardLabel);
+    var ctx=board.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(piecePosition * 29, 25, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function populateClaimedCanvases() {
+    var unclaimedBoard = document.getElementById("unclaimedCanvas");
+    var unclaimedctx = unclaimedBoard.getContext("2d");
+    unclaimedctx.clearRect(0, 0, unclaimedBoard.width, unclaimedBoard.height);
+    var claimedBoard1 = document.getElementById("player1canvas");
+    var player1ctx = claimedBoard1.getContext("2d");
+    player1ctx.clearRect(0, 0, claimedBoard1.width, claimedBoard1.height);
+    var claimedBoard2 = document.getElementById("player2canvas");
+    var player2ctx = claimedBoard2.getContext("2d");

```

Comment:
A lot of duplicate code between claimed board 1 and 2!  Same comment as above.  As a general rule, try to avoid repeating yourself!